### PR TITLE
Fix(App): Use NC components

### PR DESCRIPTION
**Pull request by DorraJaouad**
* Added  @nextcloud/dialogs to dependencies to inform user when there is an error.
* Relaced native HTML elements with NC components in App components.
* Fixed name attr. 

| Before | After |
|--|--|
|image|image|

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -21,7 +21,7 @@
   -->
 
 <template>
-	<NcSettingsSection :title="t('bruteforcesettings', 'Brute-force IP whitelist')"
+	<NcSettingsSection :name="t('bruteforcesettings', 'Brute-force IP whitelist')"
```
Comment (nickvergessen) [inline src/App.vue:24]: this should be backported iirc? Was also a problem on older versions?

Comment (DorraJaouad) [reply src/App.vue:24]: 27 no as it is still on older version but will check 28

**Comment by DorraJaouad** [comment]
I was waiting for a confirmation to nothing to change before I compile

File: src/App.vue [hunk 2]
```diff
@@ -42,39 +42,49 @@
...
+				class="whitelist__mask"
+				:value.sync="newWhitelist.mask"
 				type="number"
 				name="mask"
 				min="1"
 				max="128"
 				maxlength="2"
-				placeholder="64">
-			<NcButton native-type="submit">
+				label="Mask"
```
Comment (nickvergessen) [inline src/App.vue:62]: Followup: should be translated

File: src/App.vue [hunk 3]
```diff
@@ -42,39 +42,49 @@
...
+		<div class="whitelist__form">
+			<NcInputField id="ip"
+				class="whitelist__ip"
+				:value.sync="newWhitelist.ip"
 				type="text"
 				name="ip"
-				placeholder="2001:db8::">/
-			<input id="mask"
-				v-model="newWhitelist.mask"
+				label="IP address"
```
Comment (nickvergessen) [inline src/App.vue:52]: Followup: should be translated

**Review by nickvergessen — APPROVED** [review]
While on it change the screenshot in https://github.com/nextcloud/bruteforcesettings/tree/master/screenshots ?

**Comment by DorraJaouad** [comment]
> While on it change the screenshot in [`master`/screenshots](https://github.com/nextcloud/bruteforcesettings/tree/master/screenshots?rgh-link-date=2024-02-09T15%3A21%3A07Z) ?

Yes I will

**Pull request merged**